pkg/cortex/serve/cortex_internal/lib/storage/s3.py

```diff
@@ -146,18 +150,16 @@ def _read_bytes_from_s3_single(self, key, allow_missing=False, ext_bucket=None):
 
         return byte_array.strip()
 
-    def search(self, prefix="", suffix="") -> Tuple[List[str], List[datetime.datetime]]:
+    def search(
+        self, prefix="", suffix="", include_dir_objects=False
+    ) -> Tuple[List[str], List[datetime.datetime]]:
         paths = []
         timestamps = []
 
         timestamp_map = {}
-        for key, ts in self._get_matching_s3_keys_generator(prefix, suffix):
-            timestamp_map[key] = ts
-
-        filtered_keys = util.remove_non_empty_directory_paths(list(timestamp_map.keys()))
```

Comment:
Is the `remove_non_empty_directory_paths()` helper still used, or can it be removed?

Reply:
It's no longer required here, but it is in other parts of the code-base.